scripts/src/java/org/oppia/android/scripts/common/BazelClient.kt

```diff
@@ -130,6 +132,52 @@ class BazelClient(private val rootDirectory: File, private val commandExecutor:
     return correctedTargets
   }
 
+  /**
+   * Runs code coverage for the specified Bazel test target.
+   *
+   * @param bazelTestTarget Bazel test target for which code coverage will be run
+   * @return the coverage data file path generated from bazel coverage
+   */
+  fun runCoverageForTestTarget(bazelTestTarget: String): ByteArray? {
+    val coverageData = executeBazelCommand(
+      "coverage",
+      bazelTestTarget
+    )
+    val coverageDataFilePath = parseCoverageDataFile(coverageData)
+    val coverageDataFileContent = readDatFileAsBinary(coverageDataFilePath!!)
+    return coverageDataFileContent
```

Comment:
The nullability seems a bit off here since we can possibly return null. Perhaps something like this would be a bit more robust:

```suggestion
    return parseCoverageDataFile(coverageData)?.let { path ->
      readDatFileAsBinary(path)
    }
```

Note this means that ``readDatFileAsBinary`` can also be simplified since it never takes a null string value.

Reply:
Thanks Ben. 
I didn't realize the nullability issue there. I've now updated the code as suggested using safe calls. 

Revised versions:
**parseCoverageDataFile Changes:** https://github.com/oppia/oppia-android/pull/5423/files#diff-1e798d9f6f6e6059ff0d747cc961c02e0ba9bf4bf588dfe9db710be512d0c63dR150-R152

**readDatFile (previously readDatFileAsBinary) Changes:** https://github.com/oppia/oppia-android/pull/5423/files#diff-1e798d9f6f6e6059ff0d747cc961c02e0ba9bf4bf588dfe9db710be512d0c63dR181-R183

Reply:
Ah, I think part of my earlier comment might not have been as clear so I left a newer comment. The other aspect of this has been addressed, though, so resolving this thread.